solitude/exceptions.py

```diff
@@ -18,11 +17,8 @@ class InvalidQueryParams(ParseError):
 
 
 def custom_exception_handler(exc):
-    # We shouldn't be rolling back if we are inside a transaction test case
-    # until we figure out mozilla/solitude#388
-    if not settings.IN_TEST_SUITE:
-        log.warning('Transaction rolled back due to error.')
-        rollback()
+    if not getattr(exc, '_do_not_rollback', False):
```

Comment:
at least a log message on the else branch which is when exceptions won't cause rollbacks